haskell/upload-hackage-docs.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# SPDX-FileCopyrightText: 2020 Serokell <https://serokell.io/>
+#
+# SPDX-License-Identifier: MPL-2.0
```

Comment:
@maksbotan I don't know about the origin of this script so please check whether you are ok with this copyright and license.

Reply:
I think that I based it on https://github.com/phaazon/stack-haddock-upload. It's "BSD 0-Clause" license, so technically we are free to do whatever we want without the need to mention that repo.

And maybe there was some stuff on StackOverflow I've also used, but I can't find that now.